dataplane/slow_worker.cpp

```diff
@@ -1,25 +1,968 @@
 #include "slow_worker.h"
 
-#include "common/type.h"
-#include "controlplane.h"
-#include "worker.h"
+#include "dataplane.h"
+#include "icmp_translations.h"
 
 namespace dataplane
 {
+SlowWorker::SlowWorker(cWorker* worker,
+                       const std::vector<tPortId>& ports_to_service,
+                       const std::vector<cWorker*>& workers_to_service,
+                       std::vector<dpdk::RingConn<rte_mbuf*>>&& from_gcs,
+                       KernelInterfaceWorker&& kni,
+                       rte_mempool* mempool,
+                       bool use_kni,
+                       uint32_t sw_icmp_out_rate_limit) :
+        m_ports_serviced{ports_to_service},
+        m_workers_serviced{workers_to_service},
+        from_gcs_{from_gcs},
+        m_slow_worker{worker},
+        m_mempool{mempool},
+        m_fragmentation(
+                [this](rte_mbuf* pkt, const common::globalBase::tFlow& flow) {
+	                SendToSlowWorker(pkt, flow);
+                },
+                m_slow_worker->dataPlane->getConfigValues().fragmentation),
+        m_dregress(this,
+                   m_slow_worker->dataPlane,
+                   static_cast<uint32_t>(m_slow_worker->dataPlane->getConfigValues().gc_step)), // @TODO fix mismatch in type of config value and actually used one
+        m_config{sw_icmp_out_rate_limit, use_kni},
+        m_kni_worker{std::move(kni)}
+{
+	m_workers_serviced.emplace_back(m_slow_worker);
+}
+
+SlowWorker::SlowWorker(SlowWorker&& other) :
+        m_ports_serviced{std::move(other.m_ports_serviced)},
+        m_workers_serviced{std::move(other.m_workers_serviced)},
+        m_slow_worker{other.m_slow_worker},
+        m_mempool{other.m_mempool},
+        m_fragmentation{std::move(other.m_fragmentation)},
+        m_dregress{std::move(other.m_dregress)},
+        m_kni_worker{std::move(other.m_kni_worker)}
+{
+	m_fragmentation.Callback() = [this](rte_mbuf* pkt, const common::globalBase::tFlow& flow) {
+		SendToSlowWorker(pkt, flow);
+	};
+}
+
```

Comment:
What is the difference between this and the default move constructor? It appears we are simply moving all fields, which is the behavior of the default move constructor. Is it necessary to explicitly set the `Callback()` to `SendToSlowWorker(pkt, flow)` after the move? 
I mean ff the `Callback()` could potentially point to a different function, changing it after the move seems strange and could imply we are not just moving an object but also modifying its state

Reply:
`Callback()` lambda is capturing `this` pointer to SlowWorker. Since we are moving SlowWorker the old pointer passed in this lambda to `Fragmentation` becomes invalid after move.